pkg/reconciler/contour/resources/kingress.go

```diff
@@ -94,6 +101,10 @@ func MakeEndpointProbeIngress(ctx context.Context, ing *v1alpha1.Ingress, previo
 
 	for _, name := range l {
 		si := sns[name]
+		if si.HasPath {
```

Comment:
not very familiar with contour. Does this mean the service that has path won't be added into the route rules? Then will the http01 challenge service still be accessible and be able to serve http01 challenge traffic?

Reply:
Ha, no.  This is a *recursive* kingress that creates a host for each service that we can probe to ensure that all Envoys know about all of the Endpoints before we roll out new route programming.

We can't do this for HTTP01 challenges (the only thing we use Path for) because certmanager doesn't support proper probe responses.